src/helper.js

```diff
@@ -191,3 +193,40 @@ function mergeConfig(oldConfig, newConfig) {
 		}
 	});
 }
+
+function parseHostmask(hostmask) {
+	let nick = "";
+	let ident = "*";
+	let hostname = "*";
+	let parts = [];
+
+	// Parse hostname first, then parse the rest
+	parts = hostmask.split("@");
+
+	if (parts.length >= 2) {
+		hostname = parts[1] || "*";
+		hostmask = parts[0];
+	}
+
+	parts = hostmask.split("!");
+
+	if (parts.length >= 2) {
+		ident = parts[1] || "*";
+		hostmask = parts[0];
+	}
+
+	nick = hostmask || "*";
+
+	const result = {
```

Comment:
As you asked on IRC, hostmasks indeed also need to be compared case insensitively. I think it's pretty safe to lowercase it in parse function (instead of in compare function all the time).

Don't forget to add tests for lowercasing :)